### ISSUE TITLE
SNS unable to dispatch messages to localstack:SQS

### ISSUE THREAD

**Issue by matt-wormley** [localstack/localstack#259]
Using the below sample code, localstack is unable to deliver a message to a localstack SQS queue through SNS subscription. Is localstack SNS attempting to deliver this message to real-aws? Since only the arn is in the subscription...

sample code:
```
require 'aws-sdk-core'

sns = Aws::SNS::Client.new(endpoint: 'http://localhost:4575')
sqs = Aws::SQS::Client.new(endpoint: 'http://localhost:4576')

topic_arn = sns.create_topic(name: 'foo').topic_arn
queue_url = sqs.create_queue(queue_name: 'foo-q').queue_url
queue_arn = sqs.get_queue_attributes(queue_url: queue_url).attributes["QueueArn"]

sns.subscribe(topic_arn: topic_arn, protocol: 'sqs', endpoint: queue_arn)

sns.publish(topic_arn: topic_arn, message: 'foo-message')
```

error from localstack:
```
2017-08-24T14:54:15:ERROR:localstack.services.generic_proxy: Error forwarding request: An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the GetQueueUrl operation: The specified queue does not exist for this wsdl version. Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 166, in forward
    path=path, data=data, headers=forward_headers)
  File "/opt/code/localstack/localstack/services/sns/sns_listener.py", line 60, in forward_request
    queue_url = aws_stack.get_sqs_queue_url(queue_name)
  File "/opt/code/localstack/localstack/utils/aws/aws_stack.py", line 288, in get_sqs_queue_url
    response = client.get_queue_url(QueueName=queue_name)
  File "/opt/code/localstack/.venv/lib/python2.7/site-packages/botocore/client.py", line 253, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/opt/code/localstack/.venv/lib/python2.7/site-packages/botocore/client.py", line 557, in _make_api_call
    raise error_class(parsed_response, operation_name)
QueueDoesNotExist: An error occurred (AWS.SimpleQueueService.NonExistentQueue) when calling the GetQueueUrl operation: The specified queue does not exist for this wsdl version.
```

**Comment by matt-wormley** [comment]
I have looked into this and have figure out it has to do with how the proxy picks up credentials. In my case we use .aws/{config|certificates} profiles, and we map our ~/.aws as /root/.aws in the localhost container. However, the dockerfile sets AWS_ACCESS_KEY_ID and AWS_SECRET_KEY so that overrides the ~/.aws credentials and so we end up with `foobar` in the authentication string that is used to contact moto-sqs from our incoming sns publish API.

I worked out some fixes 1) pulling out AWS_* keys from Dockerfile and 2) sending in region_name = None into the boto client initializer and these fixes solved my initial problem. I'm having trouble getting the right mix of things, tho, for the unit tests.

**Comment by matt-wormley** [comment]
I ended up with a 1 word change... when LOCAL_REGION is in use we should send in region_name=None to boto and not DEFAULT_REGION when constructing boto client objects.

**Comment by whummer** [comment]
Thanks for digging into this @matt-wormley . So, when we set the region to `None` by default, `boto` will use its regular mechanism to figure out which region is configured in the local configuration, right? Can you please create a pull request with the fix? I have a feeling that this change may have some potential side effects (I'll dig in a bit more to find this out), but if the tests are passing with your change, I'd be happy to merge. Thanks

**Comment by matt-wormley** [comment]
I submitted the PR. It looks like the CI automation balked but I'm not sure why. Hopefully you get an idea of what I've found. I don't know if I have the right solution here, but it does solve our own issue for the time being. Thanks so much.

**Comment by whummer** [comment]
Fixed in #389
